backend-elasticsearch/src/main/java/org/hibernate/search/v6poc/backend/elasticsearch/search/predicate/impl/BooleanJunctionPredicateBuilderImpl.java

```diff
@@ -43,6 +57,34 @@ public ElasticsearchSearchPredicateCollector getFilterCollector() {
 		return this::filter;
 	}
 
+	@Override
+	public void minimumShouldMatchNumber(int constraintIgnoreCeiling, int matchingClausesNumber) {
```

Comment:
Maybe s/constraintIgnoreCeiling/applyConstraintIfHasMoreClausesThan/ .

My preference would be for `threshold` but I don't know how to convey the fact that it is strict. Maybe ask Sanne? He's generally of good advice for that.